src/mappings/core.ts

```diff
@@ -485,59 +492,39 @@ export function handleSwap(event: Swap): void {
   swap.save()
 
   // update the transaction
+
+  // TODO: Consider using .concat() for handling array updates to protect
+  // against unintended side effects for other code paths.
   swaps.push(swap.id)
   transaction.swaps = swaps
   transaction.save()
 
   // update day entities
-  updatePairDayData(event)
-  updatePairHourData(event)
-  updateUniswapDayData(event)
-  updateTokenDayData(token0 as Token, event)
-  updateTokenDayData(token1 as Token, event)
-
-  let timestamp = event.block.timestamp.toI32()
-  // daily info
-  let dayID = timestamp / 86400
-  let dayPairID = event.address
-    .toHexString()
-    .concat('-')
-    .concat(BigInt.fromI32(dayID).toString())
-
-  // hourly info
-  let hourID = timestamp / 3600
-  let hourPairID = event.address
-    .toHexString()
-    .concat('-')
-    .concat(BigInt.fromI32(hourID).toString())
```

Comment:
Instead of repeating this logic here I think it makes sense to simply re-use the entities as they are updated/created in the called functions.

Reply:
Yep good idea